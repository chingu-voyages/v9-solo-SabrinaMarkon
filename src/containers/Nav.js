import React from 'react';
import { NavLink } from 'react-router-dom';

export default function Nav() {

  return (
      <nav className="nav">
          <ul>
            <li key="0" className="nav__item">
              <NavLink className="nav__link" exact activeClassName="active" to="/">Home</NavLink>
            </li>
            <li key="1" className="nav__item">
              <NavLink className="nav__link" exact activeClassName="active" to="/howitworks">How It Works</NavLink>
            </li>
            <li key="2" className="nav__item">
              <NavLink className="nav__link" exact activeClassName="active" to="/signup">Signup</NavLink>
            </li>
            <li key="3" className="nav__item">
              <NavLink className="nav__link" exact activeClassName="active" to="/login">Login</NavLink>
            </li>
            <li key="4" className="nav__item">
              <NavLink className="nav__link" activeClassName="active" to="/postjob">Post a Job!</NavLink>
            </li>
          </ul>
      </nav>
  )
}