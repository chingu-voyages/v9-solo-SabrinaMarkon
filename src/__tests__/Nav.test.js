import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

// components
import Nav from '../containers/Nav';

let wrapper;

beforeEach(() => {
  wrapper = mount(<MemoryRouter><Nav /></MemoryRouter>);
});

describe('Testing Nav component', () => {
  it('renders the navigation menu', () => {
    const nav = document.createElement('nav');
    ReactDOM.render(wrapper, nav);
    ReactDOM.unmountComponentAtNode(nav);
  });

  describe('Test the NavLinks in the navigation menu', () => {
    it('should render 5 NavLink elements', () => {
      expect(wrapper.find('NavLink')).toHaveLength(5);
    });

    it('includes a NavLink to the Home page', function() {
      expect(wrapper.find('NavLink').at(0).prop('to')).toEqual('/');
    });

    it('includes a NavLink to the How It Works page', function() {
      expect(wrapper.find('NavLink').at(1).prop('to')).toEqual('/howitworks');
    });

    it('includes a NavLink to the Signup page', function() {
      expect(wrapper.find('NavLink').at(2).prop('to')).toEqual('/signup');
    });

    it('includes a NavLink to the Login page', function() {
      expect(wrapper.find('NavLink').at(3).prop('to')).toEqual('/login');
    });

    it('includes a NavLink to the Post a Job page', function() {
      expect(wrapper.find('NavLink').at(4).prop('to')).toEqual('/postjob');
    });
  });

  it('it should match the Nav snapshot', () => {
    // If snapshot exists, compare. If not, create the snapshot.
    expect(wrapper).toMatchSnapshot();
  });

});