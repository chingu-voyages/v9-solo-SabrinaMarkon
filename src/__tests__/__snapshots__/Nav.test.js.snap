// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Nav component it should match the Nav snapshot 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "7elljx",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "7elljx",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Nav>
      <nav
        className="nav"
      >
        <ul>
          <li
            className="nav__item"
            key="nav-ss0"
          >
            <NavLink
              activeClassName="active"
              className="nav__link"
              exact={true}
              to="/"
            >
              <Link
                aria-current="page"
                className="nav__link active"
                style={Object {}}
                to="/"
              >
                <a
                  aria-current="page"
                  className="nav__link active"
                  href="/"
                  onClick={[Function]}
                  style={Object {}}
                >
                  Home
                </a>
              </Link>
            </NavLink>
          </li>
          <li
            className="nav__item"
            key="nav-1"
          >
            <NavLink
              activeClassName="active"
              className="nav__link"
              exact={true}
              to="/howitworks"
            >
              <Link
                aria-current={null}
                className="nav__link"
                to="/howitworks"
              >
                <a
                  aria-current={null}
                  className="nav__link"
                  href="/howitworks"
                  onClick={[Function]}
                >
                  How It Works
                </a>
              </Link>
            </NavLink>
          </li>
          <li
            className="nav__item"
            key="nav-2"
          >
            <NavLink
              activeClassName="active"
              className="nav__link"
              exact={true}
              to="/signup"
            >
              <Link
                aria-current={null}
                className="nav__link"
                to="/signup"
              >
                <a
                  aria-current={null}
                  className="nav__link"
                  href="/signup"
                  onClick={[Function]}
                >
                  Signup
                </a>
              </Link>
            </NavLink>
          </li>
          <li
            className="nav__item"
            key="nav-3"
          >
            <NavLink
              activeClassName="active"
              className="nav__link"
              exact={true}
              to="/login"
            >
              <Link
                aria-current={null}
                className="nav__link"
                to="/login"
              >
                <a
                  aria-current={null}
                  className="nav__link"
                  href="/login"
                  onClick={[Function]}
                >
                  Login
                </a>
              </Link>
            </NavLink>
          </li>
          <li
            className="nav__item"
            key="nav-4"
          >
            <NavLink
              activeClassName="active"
              className="nav__link"
              to="/postjob"
            >
              <Link
                aria-current={null}
                className="nav__link"
                to="/postjob"
              >
                <a
                  aria-current={null}
                  className="nav__link"
                  href="/postjob"
                  onClick={[Function]}
                >
                  Post a Job!
                </a>
              </Link>
            </NavLink>
          </li>
        </ul>
      </nav>
    </Nav>
  </Router>
</MemoryRouter>
`;
